File: back/src/main/java/com/hp/hospin/hospital/domain/type/Hospital.java
package com.hp.hospin.hospital.domain.type;

import lombok.Getter;

@Getter
public class Hospital {
    private final String hospitalCode;
    private final String name;
    private final Long categoryCode;
    private final Long regionCode;
    private final Long districtCode;
    private final Long postalCode;
    private final String address;
    private final String callNumber;
    private final String latitude;
    private final String longitude;

    public Hospital(String hospitalCode,
                    String name,
                    Long categoryCode,
                    Long regionCode,
                    Long districtCode,
                    Long postalCode,
                    String address,
                    String callNumber,
                    String latitude,
                    String longitude) {
        this.hospitalCode = hospitalCode;
        this.name = name;
        this.categoryCode = categoryCode;
        this.regionCode = regionCode;
        this.districtCode = districtCode;
        this.postalCode = postalCode;
        this.address = address;
        this.callNumber = callNumber;
        this.latitude = latitude;
        this.longitude = longitude;
    }
}

File: front/src/lib/shared/types/hospital.ts
// --- hospital.ts (기존 types/hospital/hospital.ts) ---
export type Hospital = {
	hospital_code: string;
	name: string;
	address: string;
	latitude: number;
	longitude: number;
};

// --- info.ts (기존 types/hospital/info.ts) ---
export interface HospitalInfoResponse {
  baseInfo: BaseInfo;
  detailInfo: DetailInfo;
  gradeInfo: GradeInfo;
}

export interface BaseInfo {
  hospitalCode: string;
  name: string;
  address: string;
  callNumber: string;
  latitude: string;   // 백엔드가 string으로 주므로 주의
  longitude: string;  // 백엔드가 string으로 주므로 주의
}

export interface DetailInfo {
  departmentCodes: string[];
  closedSunday: string;
  closedHoliday: string;
  emergencyDayYn: string;
  emergencyDayPhone1: string;
  emergencyDayPhone2: string;
  emergencyNightYn: string;
  emergencyNightPhone1: string;
  emergencyNightPhone2: string;
  lunchWeekday: string;
  lunchSaturday: string;
  receptionWeekday: string;
  receptionSaturday: string;
  treatSunStart: string; treatSunEnd: string;
  treatMonStart: string; treatMonEnd: string;
  treatTueStart: string; treatTueEnd: string;
  treatWedStart: string; treatWedEnd: string;
  treatThuStart: string; treatThuEnd: string;
  treatFriStart: string; treatFriEnd: string;
  treatSatStart: string; treatSatEnd: string;
}

export interface GradeInfo {
  grades: Record<string, number>;
}

// --- list.ts (기존 types/hospital/list.ts) ---
export interface HospitalListItemRaw {
	hospitalCode?: string;
	hospital_code?: string;
	name: string;
	address: string;
	callNumber?: string | null;
}

// 프론트에서 일관되게 쓸 표준화 타입
export interface HospitalListItem {
	hospitalCode: string;
	name: string;
	address: string;
	callNumber?: string | null;
}

// 표준화 유틸: raw → normalized
export function normalizeHospitalListItem(raw: HospitalListItemRaw): HospitalListItem {
	return {
		hospitalCode: raw.hospitalCode ?? raw.hospital_code ?? '',
		name: raw.name,
		address: raw.address,
		callNumber: raw.callNumber ?? null
	};
}

// Spring Data Page 응답 최소 형태 제너릭
export interface PageResult<T> {
	content: T[];
	totalPages: number;
	totalElements: number;
	// number?: number;
	// size?: number;
	// first?: boolean;
	// last?: boolean;
}

export interface HospitalListResponse {
	hospital_code: string;
	name: string;
	address: string;
	latitude: number;
	longitude: number;
}
